workflow/controller/operator.go

```diff
@@ -2285,7 +2285,10 @@ func (woc *wfOperationCtx) executeScript(nodeName string, templateScope string,
 	}
 
 	mainCtr := tmpl.Script.Container
-	mainCtr.Args = append(mainCtr.Args, common.ExecutorScriptSourcePath)
+	// if script source is specified then pass all container args to the script
+	if tmpl.Script.Source != "" {
```

Comment:
Test?

Reply:
@alexec I didnt have time to test it properly yet, thats why I created a draft PR. But I will try to finish this today.